src/services/common/wmts.helper.ts

```diff
@@ -0,0 +1,19 @@
+import { remoteLayersService } from '@/components/remote-layers/remote-layers.service'
+import { IdValues } from '../layer-metadata/layer-metadata.model'
+
+export class WmtsHelper {
+  async getMetadata(idValues: IdValues) {
+    console.assert(idValues.serviceType === 'WMTS')
+    const wmtsEndpoint = remoteLayersService.getWmtsEndpoint(idValues.wmsUrl)
+    await wmtsEndpoint.isReady()
+    const service = wmtsEndpoint?.getServiceInfo()
+    const layer = wmtsEndpoint?.getLayerByName(idValues.layerName)
+    return {
+      title: layer.Title,
+      description: layer.Abstract,
+      accessConstraints: service.AccessConstraints,
+      serviceDescription: service.Abstract,
+    }
+  }
+}
+export const wmtsHelper = new WmtsHelper()
```

Comment:
I added an abstract `OwsHelper` that is extended by the `WmtsHelper`

Reply:
Sorry, I meant `static` for `getMetadata()` method, because I don't see why this class must be instantiate.
But this is not very a big deal. Ok for extending the abstract as well.